core/src/main/java/tc/oc/pgm/fireworks/FireworkMatchModule.java

```diff
@@ -0,0 +1,259 @@
+package tc.oc.pgm.fireworks;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import org.bukkit.Bukkit;
+import org.bukkit.Color;
+import org.bukkit.FireworkEffect;
+import org.bukkit.FireworkEffect.Type;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Firework;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.inventory.meta.FireworkMeta;
+import tc.oc.pgm.Config;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.party.Competitor;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.controlpoint.events.ControllerChangeEvent;
+import tc.oc.pgm.core.CoreLeakEvent;
+import tc.oc.pgm.destroyable.DestroyableDestroyedEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.flag.event.FlagCaptureEvent;
+import tc.oc.pgm.regions.Bounds;
+import tc.oc.pgm.regions.Region;
+import tc.oc.pgm.wool.PlayerWoolPlaceEvent;
+import tc.oc.util.bukkit.BukkitUtils;
+import tc.oc.util.bukkit.block.BlockVectors;
+
+@ListenerScope(MatchScope.LOADED)
+public class FireworkMatchModule implements MatchModule, Listener {
+
+  private final Match match;
+
+  private static final boolean POST_ENABLED = Config.Fireworks.postMatchEnabled();
+  private static final boolean GOALS_ENABLED = Config.Fireworks.goalsEnabled();
+
+  private static final int NUMBER = 5;
+  private static final int DELAY = 40;
+  private static final int FREQUENCY = 40;
+  private static final int ITERATION_COUNT = 15;
+  private static final int POWER = 2;
```

Comment:
Could use some better naming or descriptions (i'm not 100% sure if i got them all right or i mixed any of them up):
```suggestion
  private static final int MAX_ROCKET_COUNT = 5; // Maximum rockets to throw at once, one per player
  private static final int INITIAL_DELAY = 40; // Ticks before starting to throw rockets
  private static final int FREQUENCY = 40; // Ticks between rocket throws
  private static final int ITERATION_COUNT = 15; // Amount of times rockets are thrown
  private static final int ROCKET_POWER = 2; // Power of the rockets, 1 = low, 2 = medium, 3 = high
```

Reply:
Great suggestion @Pablete1234! I’ll take care of this as soon as I can, sorry for the delay.